class/grid.lua

```diff
@@ -206,4 +210,11 @@ Grid = Class {
             end
         end
     end;
+    vacateSpaces = function(self, structure)
```

Comment:
I would either give this a more explicit name, eg. `vacateSpacesForStructure` or change the signature so it takes an x, y, width and height instead, so its more generic.